### PR TITLE
fix: cleans undesirable strings from rss feeds

### PR DESCRIPTION
**Pull request by Doug-Reed**

### CODE REVIEW

File: src/main/java/edu/wisc/my/rssToJson/service/RsstoJsonServiceImpl.java [hunk 1]
```diff
@@ -51,6 +54,37 @@ public JSONObject getJsonFromURL(String endpoint) {
         }
         jsonToReturn.put("items", entries);
         jsonToReturn.put("status", "ok");
-        return jsonToReturn;
+        return getJSONCleaned(jsonToReturn);
     }
+
+    private JSONObject getJSONCleaned(JSONObject jsonToClean) {
```
Comment (apetro) [inline src/main/java/edu/wisc/my/rssToJson/service/RsstoJsonServiceImpl.java]: Suggest: reserve `get` prefix for JavaBean accessors. This method could be `cleanJson(JSONobject dirtyJSON)` or `filterJson(JSONobject input)` or `replaceProblematicCharacters(JSONObject withUnwashedCharacters)`

File: src/main/java/edu/wisc/my/rssToJson/service/RsstoJsonServiceImpl.java [hunk 2]
```diff
@@ -51,6 +54,37 @@ public JSONObject getJsonFromURL(String endpoint) {
...
+      String tellTaleHighBitIndicator = "\\u";
```
Comment (apetro) [inline src/main/java/edu/wisc/my/rssToJson/service/RsstoJsonServiceImpl.java]: What's telltale about this indicator? What tale does it tell?

File: src/main/java/edu/wisc/my/rssToJson/service/RsstoJsonServiceImpl.java (hunk 1 above)
Comment (apetro) [inline src/main/java/edu/wisc/my/rssToJson/service/RsstoJsonServiceImpl.java]: Suggest: what does this method do? Would it be better documented with a JavaDoc comment?

Comment (apetro) [inline src/main/java/edu/wisc/my/rssToJson/service/RsstoJsonServiceImpl.java]: This feels like a procedural style of programming. Private method to implement the subroutine. Consider modeling this instead as a helper object. What is this? A `JsonCleaner`? A `StringFilter`? Factoring this out as tight object(s) with tight API(s) might make it easy and fun to unit test this tight bit of responsibility.

File: src/main/java/edu/wisc/my/rssToJson/service/RsstoJsonServiceImpl.java [hunk 3]
```diff
@@ -51,6 +54,37 @@ public JSONObject getJsonFromURL(String endpoint) {
...
+      if (!jsonToClean.toString().contains(tellTaleHighBitIndicator)) {
+        return jsonToClean;
+      }
+      try {
+        Resource resource = new ClassPathResource("stringCleaner.json");
```
Comment (apetro) [inline src/main/java/edu/wisc/my/rssToJson/service/RsstoJsonServiceImpl.java]: There are two known problematic characters. Is that worth externalizing into the JSON file or would it be clearer to inline that bit of configuration in the Java?

This `try` block becomes ~2 lines of code with that inlining? So it's a tradeoff maybe worth making.

File: src/main/java/edu/wisc/my/rssToJson/service/RsstoJsonServiceImpl.java [hunk 4]
```diff
@@ -51,6 +65,20 @@ public JSONObject getJsonFromURL(String endpoint) {
         }
         jsonToReturn.put("items", entries);
         jsonToReturn.put("status", "ok");
+        //Clean undesirable unicode literals from the JsonObject
```
Comment (apetro) [inline src/main/java/edu/wisc/my/rssToJson/service/RsstoJsonServiceImpl.java:68]: Inline further for brevity?

Lines 68 through 78 reduce to
```
jsonStringToClean = jsonStringToClean.replaceAll("\\u2019", "'"); // right curly single quote becomes single quote
jsonStringToClean = jsonStringToClean.replaceAll("\\u2014", "-"); // en dash becomes minus sign
```

thereby eliminating need for `unicodeToClean`?

Like, the `replaceAll`s are two lines of code, they don't need to be represented as JSON?

File: src/main/java/edu/wisc/my/rssToJson/service/RsstoJsonServiceImpl.java [hunk 5]
```diff
@@ -20,6 +23,17 @@
 
     private RssToJsonDao rssToJsonDao;
 
+    // JsonArray of unicode characters
+    // offender is the unicode character we're replacing
+    // replacement is the ASCII character we're subbing in.
+    private String unicodeToClean = "{\"replacements\":[{" +
```
Comment (apetro) [inline src/main/java/edu/wisc/my/rssToJson/service/RsstoJsonServiceImpl.java:29]: I'd have named this something more like "characterMappings". This String isn't going to be itself cleaned so the "toClean" isn't doing it for me -- I don't think of my mop as my "mopToClean" either. :)

File: src/main/java/edu/wisc/my/rssToJson/service/RsstoJsonServiceImpl.java [hunk 6]
```diff
@@ -20,6 +23,17 @@
...
+      "\"offender\":" + "\"\\\\" + "u2019" + "\"," + //right curly single quote
```
Comment (apetro) [inline src/main/java/edu/wisc/my/rssToJson/service/RsstoJsonServiceImpl.java:30]: left curly single quote isn't also a problem?

**Comment by paulerickson** [comment]
What's so undesirable about these characters?  I'm rather fond of them!

**Pull request closed without merging**